packages/contracts/contracts/voting/erc20/Erc20VotingManager.sol

```diff
@@ -0,0 +1,215 @@
+                abi.encodeWithSelector(
+                    GovernanceWrappedERC20.initialize.selector,
+                    IERC20Upgradeable(_tokenSetting.addr),
+                    _tokenSetting.name,
+                    _tokenSetting.symbol
+                )
+            );
+        }
+
+        bytes memory init = abi.encodeWithSelector(
+            ERC20Voting.initialize.selector,
+            _dao,
+            _dao.getTrustedForwarder(),
+            _voteSetting.minTurnout,
+            _voteSetting.minSupport,
+            _voteSetting.minDuration,
+            GovernanceERC20(_tokenSetting.addr)
+        );
+
+        plugin = createProxy(dao, getImplementationAddress(), init);
+    }
+
+    /// @inheritdoc PluginManager
+    function getInstallPermissions(bytes memory data)
+        external
+        view
+        virtual
+        override
+        returns (RequestedPermission[] memory permissions, string[] memory helperNames)
+    {
+        (, TokenSetting memory _tokenSetting) = abi.decode(data, (VoteSetting, TokenSetting));
+
+        permissions = new RequestedPermission[](_tokenSetting.addr == address(0) ? 5 : 4);
+        helperNames = new string[](_tokenSetting.addr == address(0) ? 1 : 0);
+
+        address NO_ORACLE = address(0);
+
+        // Allows plugin to call DAO with EXEC_PERMISSION
+        permissions[0] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            DAO_PLACEHOLDER,
+            PLUGIN_PLACEHOLDER,
+            NO_ORACLE,
+            keccak256("EXECUTE_PERMISSION")
+        );
+
+        // Allows DAO to call plugin with SET_CONFIGURATION_PERMISSION
+        permissions[1] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            erc20VotingBase.SET_CONFIGURATION_PERMISSION_ID()
+        );
+
+        // Allows DAO to call plugin with UPGRADE_PERMISSION
+        permissions[2] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            erc20VotingBase.UPGRADE_PERMISSION_ID()
+        );
+
+        // Allows DAO to call plugin with SET_TRUSTED_FORWARDER_PERMISSION
+        permissions[3] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            erc20VotingBase.SET_TRUSTED_FORWARDER_PERMISSION_ID()
+        );
+
+        uint256 GOVERNANCE_ERC20_HELPER_IDX = 0;
+
+        if (_tokenSetting.addr == address(0)) {
+            // Allows DAO to call relatedContracts (token) with MINT_PERMISSION.
+            permissions[4] = buildPermission(
+                BulkPermissionsLib.Operation.Grant,
+                GOVERNANCE_ERC20_HELPER_IDX,
+                DAO_PLACEHOLDER,
+                NO_ORACLE,
+                keccak256("MINT_PERMISSION")
```

Comment:
It would be nice if we could reference the permission ID constants rather than duplicating them and maintaining them separately.
For example, we could put them in a library (and using them in the logic and manager contract of the plugin).
Unfortunately, all the constants in the library would be copied in the PluginManager contract (which would be wasteful in case of the DAO contract, for example).

However, an idea to think about. You can close this if you don't like it.

Reply:
yah , feel free about creating a task for it,